### ISSUE TITLE
QGIS 3.32.1 crash when opening project with missing assets (macOS)

### ISSUE THREAD

**Issue by alanterra** [qgis/QGIS#54219]
### What is the bug or the crash?

This seems to be repeatable, every time I open a project (the same project) when an asset is missing, QGIS 3.32.1 (Lima) crashes. I can always solve the problem by opening the project in QGIS 3.28 LTR, finding and updating the missing asset, saving, and then opening the project in QGIS 3.32.1. This involves a change in file format when saving in the old and new versions, and I don't know if this is related to the problem or not. I am willing to do some testing if asked.
Important: QGIS 3.28 LTR is the Mac Intel version, downloaded from qgis.org; QGIS 3.32.1 is the Mac Silicon version, built under MacPorts.

### Steps to reproduce the issue

Take an asset (like a shape file) referenced by the QGIS document and move it to a different folder. Launch QGIS 3.32.1 and open the QGIS document (I usually use the Recent Documents button on the splash screen) — immediate crash. Now open the document in QGIS 3.28 LTR, locate the asset, update, save, and quit. Now the document can be opened in QGIS 3.32.1.

### Versions

QGIS version
3.32.1-Lima
QGIS code branch
Release 3.32
Qt version
5.15.10
Python version
3.11.4
GDAL/OGR version
3.7.1
PROJ version
9.2.1
EPSG Registry database version
v10.088 (2023-05-13)
GEOS version
3.12.0-CAPI-1.18.0
SQLite version
3.42.0
PDAL version
2.5.5
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.3
OS version
macOS 13.5

Active Python plugins
pointsamplingtool
0.5.4
processing
2.12.99
grassprovider
2.12.99
db_manager
0.1.20
MetaSearch
0.3.6

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

I am quite willing (and able) to help figure out some parameters of this bug (is this a version issue or a MacSilicon vs. Intel issue?, etc.) if requested. I can share the QGIS document, but it links to about 30 GB of assets.

**Comment by agiudiceandrea** [comment]
@alanterra, thanks for reporting. Does the issue occur also using a new QGIS user profile? If yes, could you please provide a minimal sample project with only one missing layer for which the issue does occur on your system? Could you please provide a crash report, if any?

**Comment by alanterra** [comment]
Hi @agiudiceandrea

I'm sorry, I tried to replicate this with a new qgz file and it did not occur. I tried to simplify the qgz file that it is occurring in, and when simplified, the bug disappeared. I did confirm that it is still present on my machine, but I don't have the time right now to take layers out of the file until I find a simple situation that can be replicated.

All I know is that with a certain qgz file, if any of the files for a layer are missing, QGIS 3.32.1 (macOS, Apple Silicon) crashes, but as I simplify that file by removing layers, the bug disappears.

Sorry. Hopefully someone else will find this, or it will get fixed when another bug is smashed.
